src/shared.ts

```diff
@@ -1,6 +1,10 @@
+import * as vscode from 'vscode'
+
 export interface SimpleVirtualDocument {
     fullText: string
     lineText: string
+    position: vscode.Position
     offset: number
     startLine: number
+    getWordRangeAtPosition: (position: vscode.Position, regex?: RegExp | undefined) => vscode.Range | undefined
```

Comment:
Why don't we pass range instead? it's ineffective to call this function for every completion.